PWGHF/TableProducer/candidateCreatorSigmac0plusplusCascade.cxx

```diff
@@ -0,0 +1,324 @@
+        registry.fill(HIST("lc/hPtBach"), ptProng0);
+        registry.fill(HIST("lc/hPtBachVsPtCand"), ptProng0, ptCand);
+        registry.fill(HIST("lc/hPtV0"), ptProng1);
+        registry.fill(HIST("lc/hPtV0VsPtCand"), ptProng1, ptCand);
+        registry.fill(HIST("lc/hd0Bach"), impactParameter0);
+        registry.fill(HIST("lc/hd0BachVsPtCand"), impactParameter0, ptCand);
+        registry.fill(HIST("lc/hd0V0"), impactParameter1);
+        registry.fill(HIST("lc/hd0V0VsPtCand"), impactParameter1, ptCand);
+        registry.fill(HIST("lc/hd0V0pos"), dcaPosToPV);
+        registry.fill(HIST("lc/hd0V0posVsPtCand"), dcaPosToPV, ptCand);
+        registry.fill(HIST("lc/hd0V0neg"), dcaNegToPV);
+        registry.fill(HIST("lc/hd0V0negVsPtCand"), dcaNegToPV, ptCand);
+        registry.fill(HIST("lc/hPtV0pos"), ptV0Pos);
+        registry.fill(HIST("lc/hPtV0posVsPtCand"), ptV0Pos, ptCand);
+        registry.fill(HIST("lc/hPtV0neg"), ptV0Neg);
+        registry.fill(HIST("lc/hPtV0negVsPtCand"), ptV0Neg, ptCand);
+        registry.fill(HIST("lc/hV0CPA"), v0CosPA);
+        registry.fill(HIST("lc/hV0CPAVsPtCand"), v0CosPA, ptCand);
+        registry.fill(HIST("lc/hV0Radius"), v0Radius);
+        registry.fill(HIST("lc/hV0RadiusVsPtCand"), v0Radius, ptCand);
+        registry.fill(HIST("lc/hV0DCADaughters"), dcaV0Daughters);
+        registry.fill(HIST("lc/hV0DCADaughtersVsPtCand"), dcaV0Daughters, ptCand);
+        registry.fill(HIST("lc/hV0MK0Short"), mK0Short);
+        registry.fill(HIST("lc/hV0MK0ShortVsPtCand"), mK0Short, ptCand);
+        registry.fill(HIST("lc/hV0MLambda"), mLambda);
+        registry.fill(HIST("lc/hV0MLambdaVsPtCand"), mLambda, ptCand);
+        registry.fill(HIST("lc/hV0MAntiLambda"), mAntiLambda);
+        registry.fill(HIST("lc/hV0MAntiLambdaVsPtCand"), mAntiLambda, ptCand);
+        registry.fill(HIST("lc/hV0MGamma"), mGamma);
+        registry.fill(HIST("lc/hV0MGammaVsPtCand"), mGamma, ptCand);
+        registry.fill(HIST("lc/hCtV0K0Short"), ctV0K0Short);
+        registry.fill(HIST("lc/hCtV0K0ShortVsPtCand"), ctV0K0Short, ptCand);
+        registry.fill(HIST("lc/hCtV0Lambda"), ctV0Lambda);
+        registry.fill(HIST("lc/hCtV0LambdaVsPtCand"), ctV0Lambda, ptCand);
+        registry.fill(HIST("lc/hCPACand"), cpa);
+        registry.fill(HIST("lc/hCPACandVsPtCand"), cpa, ptCand);
+        registry.fill(HIST("lc/hCPAxyCand"), cpaXY);
+        registry.fill(HIST("lc/hCPAxyCandVsPtCand"), cpaXY, ptCand);
+        registry.fill(HIST("lc/hDecLengthCand"), decayLength);
+        registry.fill(HIST("lc/hDecLengthCandVsPtCand"), decayLength, ptCand);
+        registry.fill(HIST("lc/hDecLengthXYCand"), decayLengthXY);
+        registry.fill(HIST("lc/hDecLengthXYCandVsPtCand"), decayLengthXY, ptCand);
+        registry.fill(HIST("lc/hCtCand"), ctLc);
+        registry.fill(HIST("lc/hCtCandVsPtCand"), ctLc, ptCand);
+      }
+      for (const auto& trackSoftPi : tracksInThisCollision) {
+
```

Comment:
An important piece missing here is to check whether the track that you use as soft pion candidate was not already used as a daughter of your Lc.

Reply:
For softpion, we use the DCA cuts which should enable us to separate it from the weak K0s decay. Indeed, in future, I can cross-check if there is any contribution.

Reply:
true that the cut can help separating the contributions, but this does not grant a priory that the soft pion track was not used for the creation of the K0s candidate or as bachelor proton. You have to check that the track index of the soft pion is different from those of the candidate bachelor proton and the candidate K0s daughters, otherwise you risk to have candidates where a tracks is used twice.

Please fix it

Reply:
Thank you. Done.

Reply:
thaks. You still miss the same check with the bachelor proton

Reply:
Thank you. It's now done. Have a nice day!